### PR TITLE
[Issue #446] Auto hyperlink detection

### PR DESCRIPTION
**Pull request by peterphan1996**
## Status
**READY**

## Issue links
https://github.com/venturemark/webclient/issues/446

## Description
Auto hyperlink detection

## Screen Recording
https://user-images.githubusercontent.com/28189578/145206117-faecf2ba-bbd4-4267-af50-57bb7ce63493.mov

### CODE REVIEW

File: src/component/editor/element/index.tsx [hunk 1]
```diff
@@ -17,7 +18,30 @@ export function Element({ attributes, children, element }: ElementProps) {
...
+const LinkComponent = ({ attributes, children, element }: ElementProps) => {
+  const selected = useSelected();
+  return (
+    <a
+      {...attributes}
+      href={(element as LinkElement).url}
+      style={{
+        textDecoration: "none",
+        color: "blue",
+        border: selected ? "1px solid #00bcd4" : "none",
```
Comment (marcusellison) [inline src/component/editor/element/index.tsx:37]: I think there's a way to reference Plasmic style tokens in our code. Could you look into this?

Could you change the color to our teal color?

Also - could you remove the border style?

Comment (peterphan1996) [reply src/component/editor/element/index.tsx:37]: hey @marcusellison, I just changed the color to ventureGreen

**Review by tfussell — CHANGES_REQUESTED** [review]
I just tried this out. I didn't realize there was a `LinkPlugin`. Works pretty well!

My main problem is that when I'm typing a URL and press `<spacebar>`, I would expect the link to end. To reproduce, type the following `Check out https://google.com, my new website.`. The link becomes `https://google.com, my new website.` when I would expect it to just be `https://google.com`.

**Comment by peterphan1996** [comment]
> I just tried this out. I didn't realize there was a `LinkPlugin`. Works pretty well!
> 
> My main problem is that when I'm typing a URL and press `<spacebar>`, I would expect the link to end. To reproduce, type the following `Check out https://google.com, my new website.`. The link becomes `https://google.com, my new website.` when I would expect it to just be `https://google.com`.

That has to be the default of LinkPlugin, the behavior should look like this https://www.slatejs.org/examples/inlines (you can try it in here). The thing is you would have to press go right button `>` to escape the url edit mode. Let me know what you think

**Comment by peterphan1996** [comment]
New update: I see that the owner of slate-plugin has deprecated the library, the new one is Plate (ref: https://plate.udecode.io/docs/plugins/link#source-code). This one seems not to have the space-bar issue. But we have to rewrite everything with the editor using Plate

**Comment by tfussell** [comment]
> New update: I see that the owner of slate-plugin has deprecated the library, the new one is Plate (ref: https://plate.udecode.io/docs/plugins/link#source-code). This one seems not to have the space-bar issue. But we have to rewrite everything with the editor using Plate

We should probably move to Plate at some point, but let's focus on making this work if we can and keep that as a to-do. I'll try to add some code that exits the hyperlink when pressing spacebar so we can get this merged.

**Comment by tfussell** [comment]
I added:
```diff
diff --git a/src/component/editor/index.tsx b/src/component/editor/index.tsx
index 6278435..480783a 100644
--- a/src/component/editor/index.tsx
+++ b/src/component/editor/index.tsx
@@ -222,8 +222,10 @@ export function ComposeEditor({
         Transforms.move(editor, { unit: "offset", reverse: true });
         return;
       }
-      if (isKeyHotkey("right", nativeEvent)) {
-        event.preventDefault();
+      if (isKeyHotkey("right", nativeEvent) || nativeEvent.key === " ") {
+        if (nativeEvent.key !== " ") {
+          event.preventDefault();
+        }
         Transforms.move(editor, { unit: "offset" });
         return;
       }
```
and got somewhat better behavior:

https://user-images.githubusercontent.com/1735211/145914792-a1a2b9a1-04e3-4fb6-879f-656f58ba8267.mov

It requires two spaces instead of one, but this is improved over requiring pressing the right arrow. What do you think?

**Comment by peterphan1996** [comment]
@tfussell I think this is better, maybe we can merge this for now and get back to use Plate at some point in the future

**Review by tfussell — APPROVED** [review]

**Pull request merged**